Circular_kernel_3/include/CGAL/Circular_kernel_3/function_objects_polynomial_sphere.h

```diff
@@ -1168,44 +1159,17 @@ template < class SK > \
 
   public:
 
-    template <typename>
-    struct result;
-
-    // the binary overload always goes to Linear::Intersect_3
-    template <typename F, typename A, typename B>
-    struct result<F(A, B)>
-    { typedef typename Intersection_traits<SK, A, B>::result_type type; };
-
-    // This one is only for the spherical kernel, O is an output iterator
-    template <typename F, typename A, typename B, typename OutputIterator>
-    struct result<F(A, B, OutputIterator)>
-    { typedef OutputIterator type;};
-
-    // there is no quaternary form in the linear Kernel
-    template <typename F, typename A, typename B, typename C, typename OutputIterator>
-    struct result<F(A, B, C, OutputIterator)>
-    { typedef OutputIterator type; };
-
-    //only ternary from the linear kernel
-    template<typename F>
-    struct result<F(Plane_3, Plane_3, Plane_3)> {
-      typedef boost::optional<
-        boost::variant< Point_3,
-                        Line_3,
-                        Plane_3 > > type;
-    };
-
     //using SK::Linear_kernel::Intersect_3::operator();
 
     typedef typename SK::Linear_kernel::Intersect_3 Intersect_linear_3;
 
     template<class A, class B>
-    typename Intersection_traits<SK, A, B>::result_type
```

Comment:
Do you plan to deprecate `Intersection_traits` as well as usages of result_of?

Reply:
Not sure yet. I have not yet managed to remove all uses of `result`, but I will try to remove as much as possible of stuff that is useless.